### ISSUE TITLE
Config cleanup: separate experiment profiles from RL algorithm settings

### ISSUE THREAD

**Issue by xcz0** [xcz0/Eco-AutoDrive#26]
Parent: #25

## Goal

Remove the current `configs/train/` vs `configs/training/` ambiguity and establish a clear distinction between complete runnable experiment profiles and reusable RL algorithm settings.

## Current problem

The repository currently contains both:

```text
configs/train/
  ppo_smoke.yaml
  ppo_closed_loop_smoke.yaml

configs/training/
  ppo_smoke.yaml
```

The two directories encode different concepts:

- `configs/train/*` contains PPO optimizer/algorithm hyperparameters such as `gamma`, `gae_lambda`, `clip_epsilon`, learning rate, epochs, and batch sizes.
- `configs/training/ppo_smoke.yaml` is a complete runnable experiment profile composing environment, model, sampler, guidance, policy, reward, runtime, scenarios, and training settings.

This makes overrides and file discovery unnecessarily ambiguous.

## Proposed structure

Move toward:

```text
configs/
├── experiment/
│   └── train_ppo_smoke.yaml
└── rl/
    └── ppo.yaml
```

Exact filenames may be adjusted during implementation, but the semantic boundary should remain:

- `experiment/`: complete runnable profiles;
- `rl/`: reusable PPO/algorithm settings.

## Tasks

- [ ] Rename/migrate the current complete training profiles into `configs/experiment/`.
- [ ] Rename/migrate PPO hyperparameter configs from `configs/train/` into `configs/rl/`.
- [ ] Update Hydra defaults and override paths accordingly.
- [ ] Update `scripts/train.py` default `config_name`.
- [ ] Remove obsolete `configs/train/` and `configs/training/` directories once all references are migrated.
- [ ] Verify existing smoke training still resolves to the same effective algorithm/environment values.

## Acceptance criteria

- [ ] `train/` and `training/` no longer coexist as separate config concepts.
- [ ] A reader can identify runnable training experiments without inspecting YAML contents.
- [ ] PPO hyperparameters remain independently reusable by multiple future training experiments.
- [ ] Existing training smoke configuration composes successfully after migration.

## Non-goals

- Do not change PPO semantics or hyperparameter values as part of this issue.
- Do not reorganize sampler/guidance/model groups here unless required only to keep paths valid.

**Comment by xcz0** [comment]
Completed locally: moved the runnable PPO smoke profile to configs/experiment/train_ppo_smoke.yaml and reusable PPO settings to configs/rl/. Updated Hydra composition, training entrypoint, typed RL config boundary, and tests. Verified with .venv\\Scripts\\pytest.exe tests\\unit\\rl\\test_training_config.py tests\\unit\\rl\\test_ppo_config.py (12 passed) and targeted Ruff check.
